src/tools/kata-ctl/src/ops/volume_ops.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2022 Boston University
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::args::{DirectVolSubcommand, DirectVolumeCommand};
+
+use anyhow::{anyhow, Ok, Result};
+use futures::executor;
+use kata_types::mount::{
+    DirectVolumeMountInfo, KATA_DIRECT_VOLUME_ROOT_PATH, KATA_MOUNT_INFO_FILE_NAME,
+};
+use nix;
+use reqwest::StatusCode;
+use safe_path;
+use std::{fs, path::PathBuf, time::Duration};
+use url;
+
+use agent::ResizeVolumeRequest;
+use runtimes;
+
+const TIMEOUT: Duration = Duration::from_millis(200);
+const CONTENT_TYPE_JSON: &str = "application/json";
+
+pub fn handle_direct_volume(vol_cmd: DirectVolumeCommand) -> Result<()> {
+    if !nix::unistd::Uid::effective().is_root() {
+        return Err(anyhow!(
+            "super-user privileges are required for the direct-volume subcommand"
+        ));
+    }
+    let command = vol_cmd.directvol_cmd;
+    let cmd_result: Option<String> = match command {
+        DirectVolSubcommand::Add(args) => add(&args.volume_path, &args.mount_info)?,
+        DirectVolSubcommand::Remove(args) => remove(&args.volume_path)?,
+        DirectVolSubcommand::Stats(args) => executor::block_on(stats(&args.volume_path))?,
+        DirectVolSubcommand::Resize(args) => {
+            executor::block_on(resize(&args.volume_path, args.resize_size))?
+        }
+    };
+    if let Some(cmd_result) = cmd_result {
+        println!("{:?}", cmd_result);
+    }
+
+    Ok(())
+}
+
+async fn resize(volume_path: &str, size: u64) -> Result<Option<String>> {
+    let sandbox_id = get_sandbox_id_for_volume(volume_path)?;
+    let mount_info = get_volume_mount_info(volume_path)?;
+    let resize_req = ResizeVolumeRequest {
+        size,
+        volume_guest_path: mount_info.device,
+    };
+    let encoded = serde_json::to_string(&resize_req)?;
+    let shim_client = runtimes::MgmtClient::new(sandbox_id, Some(TIMEOUT))?;
+
+    let url = runtimes::DIRECT_VOLUMN_RESIZE_URL;
+    let response = shim_client
+        .post(url, &String::from(CONTENT_TYPE_JSON), &encoded)
+        .await?;
```

Comment:
1. You should use context when using a question mark.
2. I am not seeing the reason why you are using a `Result<Option<String>>` rather than simply a `Result<String>`

Reply:
AFAIC, even `Result<()>` is fine if you do not want to return any error message.